wa/framework/output.py

```diff
@@ -410,6 +413,17 @@ def get_artifact(self, name):
                 return artifact
         raise HostError('Artifact "{}" not found'.format(name))
 
+    def add_classifier(self, name, value, overwrite=False):
+        if name in self.classifiers and not overwrite:
+            raise ValueError('Cannot overwrite "{}" classifier.'.format(name))
+        self.classifiers[name] = value
+
+        for metric in self.metrics:
+            metric.classifiers[name] = value
+
+        for artifact in self.artifacts:
+            artifact.classifiers[name] = value
```

Comment:
Is it worth checking to see if the classifier already exists for any metrics or artifacts?

Reply:
I considered that. However, the way those classifiers would typically get there is form the job, so the `overwrite` check above would already handle them.

The other way, is that if the plugin itself adds them when adding the metric, but then the plugin author is responsible for both, and I figured it's better to let them sort it out at plugin implementation level, as here, we can't tell the difference between this case and the first.

Reply:
Actually, scratch that. It does't really matter. We _should_ check for clashes here.